File: 0013-roman-to-integer/0013-roman-to-integer.py
class Solution:
  def romanToInt(self, s: str) -> int:
    sum = 0
    i = 0
    dict = {
      'CM': 900, 'CD': 400, 'XC': 90, 'XL': 40, 'IX': 9, 'IV': 4,
      'M': 1000, 'D': 500, 'C': 100, 'L': 50, 'X': 10, 'V': 5, 'I': 1,
    }
    for i in range(0, len(str), 2):
      if s[i:i + 1] in dict:
        sum += s[i:i + 1]
      else:
        break
    print(sum, i)
    return sum

File: 0013-roman-to-integer/0013-roman-to-integer.php
class Solution {
  function romanToInt($s) {
    $ans = 0;
    $map = [ 'I' => 1, 'V' => 5, 'X' => 10, 'L' => 50, 'C' => 100, 'D' => 500, 'M' => 1000 ];
    for ($i = 0; $i < strlen($s); $i++) {
      if ($i && $map[$s[$i]] > $map[$s[$i - 1]]) {
        $ans -= $map[$s[$i - 1]];
        $ans += $map[$s[$i]] - $map[$s[$i - 1]];
      } else {
        $ans += $map[$s[$i]];
      }
    }
    return $ans;
  }
}